src/pip/_internal/legacy_resolve.py

```diff
@@ -125,6 +125,7 @@ def __init__(
         force_reinstall,  # type: bool
         upgrade_strategy,  # type: str
         py_version_info=None,  # type: Optional[Tuple[int, ...]]
+        require_hashes=False,  # type: Optional[bool]
```

Comment:
Is `None` actually possible here? In what circumstances?

Reply:
Nice catch, the `Optional` has been removed [here](https://github.com/pypa/pip/pull/7068/files#diff-a1f5543ad6553ad10d961f0234e3b51bR129).